### PR TITLE
T181273 Make text boxes react faster

### PR DESCRIPTION
**Pull request by gbirke**
Add wrapper function for text components that calls the change and
validation handler function on keypress events.

### CODE REVIEW

**Comment by gbirke** [comment]
This solves a problem that was discovered during investigating https://phabricator.wikimedia.org/T181277, but the description there hints more at a problem with overlapping, unclickable elements, probably caused by https://phabricator.wikimedia.org/T181273

File: skins/cat17/src/app/lib/form_components.js [hunk 1]
```diff
@@ -200,6 +201,16 @@ module.exports = {
 		bankDataElements.accountNumberElement.on( 'change', createDefaultChangeHandler( store, 'accountNumber' ) );
 		bankDataElements.bankCodeElement.on( 'change', createDefaultChangeHandler( store, 'bankCode' ) );
 		return component;
+	},
+
+	makeTextComponentMoreSnappy: function( textComponent, debounceDelay ) {
```
Comment (wiese) [inline skins/cat17/src/app/lib/form_components.js]: I get the point, but maybe `createEagerlyValidatingTextComponent` would be a better fit.

File: skins/cat17/src/app/tests/test_form_components.js [hunk 2]
```diff
@@ -320,3 +320,29 @@ test( 'Bank data component checks if all elements in the configuration are set',
...
+
+	component.element.on.withArgs( 'keypress' ).args[ 0 ][ 1 ]( fakeEvent );
+
+	setTimeout( function () {
+		t.ok( component.onChange.calledOnce, 'change handler was called once' );
+		t.ok( component.onChange.calledWith( fakeEvent ), 'change handler was called with event' );
+
+		t.ok( component.validator.calledOnce, 'validator  was called once' );
+		t.ok( component.validator.calledWith( fakeEvent ), 'validator was called with event' );
+	}, 5 );
```
Comment (wiese) [inline skins/cat17/src/app/tests/test_form_components.js]: This plays with the hope that 1 < 5 and thus the debunce will always be finished before we test?!
Wouldn't it be nicer to stub out the `_.debounce` (it's not SUT) and perform a synchronous test?
For testing the integration with lodash (if that was the goal) we could [grab a hold of the timer functions](https://lodash.com/docs#runInContext).

File: skins/cat17/src/app/lib/form_components.js [hunk 3]
```diff
@@ -1,9 +1,21 @@
...
 	NumericInputHandler = require( './numeric_input_handler' ),
 
+	/**
+	 * Wrapper around underscore debounce function with a delay of 300 milliseconds
+	 *
+	 * @param {function} f
+	 * @param {Number} milliseconds
+	 * @return {Function}
+	 */
+	defaultDebounce = function ( f, milliseconds ) {
```
Comment (JeroenDeDauw) [inline skins/cat17/src/app/lib/form_components.js]: This is never used?

**Pull request merged**